x/oracle/keeper/result.go

```diff
@@ -113,24 +81,65 @@ func (k Keeper) ResolveExpired(ctx sdk.Context, id types.RequestID) {
 // SaveResult saves the result packets for the request with the given resolve status and result.
 func (k Keeper) SaveResult(
 	ctx sdk.Context, id types.RequestID, status types.ResolveStatus, result []byte,
-) (types.OracleRequestPacketData, types.OracleResponsePacketData) {
+) error {
 	r := k.MustGetRequest(ctx, id)
-	reqPacket := types.NewOracleRequestPacketData(
-		r.ClientID,                         // ClientID
-		r.OracleScriptID,                   // OracleScriptID
-		r.Calldata,                         // Calldata
-		uint64(len(r.RequestedValidators)), // AskCount
-		r.MinCount,                         // Mincount
-	)
-	resPacket := types.NewOracleResponsePacketData(
-		r.ClientID,                // ClientID
-		id,                        // RequestID
-		k.GetReportCount(ctx, id), // AnsCount
-		int64(r.RequestTime),      // RequestTime
-		ctx.BlockTime().Unix(),    // ResolveTime
-		status,                    // ResolveStatus
-		result,                    // Result
-	)
-	k.SetResult(ctx, id, types.NewResult(reqPacket, resPacket))
-	return reqPacket, resPacket
+	reportCount := k.GetReportCount(ctx, id)
+	k.SetResult(ctx, id, types.NewResult(
+		r.ClientID,
+		r.OracleScriptID,
+		r.Calldata,
+		uint64(len(r.RequestedValidators)),
+		r.MinCount,
+		id,
+		reportCount,
+		int64(r.RequestTime),   // RequestTime
+		ctx.BlockTime().Unix(), // ResolveTime
+		status,                 // ResolveStatus
+		result,                 // Result
+	))
+
+	if r.IBCSource != nil {
+		sourceChannel := r.IBCSource.SourceChannel
+		sourcePort := r.IBCSource.SourcePort
+		sourceChannelEnd, found := k.channelKeeper.GetChannel(ctx, sourcePort, sourceChannel)
+		if !found {
+			return sdkerrors.Wrapf(channeltypes.ErrChannelNotFound, "port ID (%s) channel ID (%s)", sourcePort, sourceChannel)
+		}
+		destinationPort := sourceChannelEnd.Counterparty.PortId
+		destinationChannel := sourceChannelEnd.Counterparty.ChannelId
+		sequence, found := k.channelKeeper.GetNextSequenceSend(
+			ctx, sourcePort, sourceChannel,
+		)
+		if !found {
```

Comment:
Every packet must have a sequence number from source chain, but the channel have 2 type ORDER that mean it will reject a new packet if the sequence is not consecutive, but UNORDER is allow to process packet that not in order.